src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+import { getArticleById } from "../api";
+import { useParams } from "react-router-dom";
+import SingleArticleCard from "./SingleArticleCard";
+
+function SingleArticle() {
+  const [article, setArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const { article_id } = useParams();
+
```

Comment:
Nice use of the URL params to conditionally fetch the correct article and provide a bookmarkable link